pkg/bbgo/activeorderbook.go

```diff
@@ -92,60 +92,49 @@ func (b *ActiveOrderBook) waitAllClear(ctx context.Context, waitTime, timeout ti
 	}
 }
 
-// Cancel cancels the given order from activeOrderBook gracefully
-func (b *ActiveOrderBook) Cancel(ctx context.Context, ex types.Exchange, order types.Order) error {
-	if !b.Exists(order) {
-		return fmt.Errorf("cannot find %v in orderbook", order)
+// Cancel orders without confirmation
+func (b *ActiveOrderBook) CancelNoWait(ctx context.Context, ex types.Exchange, orders ...types.Order) error {
+	// if no orders are given, set to cancelAll
+	if len(orders) == 0 {
+		orders = b.Orders()
+	} else {
+		// simple check on given input
+		for _, o := range orders {
+			if o.Symbol != b.Symbol {
+				return errors.New("[ActiveOrderBook] cancel " + b.Symbol + " orderbook with different symbol: " + o.Symbol)
+			}
+		}
 	}
 	// optimize order cancel for back-testing
 	if IsBackTesting {
-		return ex.CancelOrders(context.Background(), order)
+		return ex.CancelOrders(context.Background(), orders...)
 	}
-	log.Debugf("[ActiveOrderBook] gracefully cancelling %v order...", order.OrderID)
-	waitTime := CancelOrderWaitTime
-
-	startTime := time.Now()
-	// ensure order is cancelled
-	for {
-		// Some orders in the variable are not created on the server side yet,
-		// If we cancel these orders directly, we will get an unsent order error
-		// We wait here for a while for server to create these orders.
-		// time.Sleep(SentOrderWaitTime)
-
-		// since ctx might be canceled, we should use background context here
-
-		if err := ex.CancelOrders(context.Background(), order); err != nil {
-			log.WithError(err).Errorf("[ActiveORderBook] can not cancel %v order", order.OrderID)
-		}
-		log.Debugf("[ActiveOrderBook] waiting %s for %v order to be cancelled...", waitTime, order.OrderID)
-		clear, err := b.waitClear(ctx, order, waitTime, 5*time.Second)
-		if clear || err != nil {
-			break
-		}
-		b.Print()
-
-		openOrders, err := ex.QueryOpenOrders(ctx, order.Symbol)
-		if err != nil {
-			log.WithError(err).Errorf("can not query %s open orders", order.Symbol)
-			continue
-		}
-
-		openOrderStore := NewOrderStore(order.Symbol)
-		openOrderStore.Add(openOrders...)
-		// if it's not on the order book (open orders), we should remove it from our local side
-		if !openOrderStore.Exists(order.OrderID) {
-			b.Remove(order)
-		}
+	log.Debugf("[ActiveOrderBook] no wait cancelling %s orders...", b.Symbol)
+	// since ctx might be canceled, we should use background context here
+	if err := ex.CancelOrders(context.Background(), orders...); err != nil {
+		log.WithError(err).Errorf("[ActiveOrderBook] no wait can not cancel %s orders", b.Symbol)
+	}
+	for _, o := range orders {
+		b.Remove(o)
```

Comment:
This could be dangerous, because the cancel RESTful API call does not guarantee the order will be executed cancel.

Reply:
If we're sure to do this, please add my comment to the code, thanks!